src/runtimes/docker/utils/normalizeContainerOS.ts

```diff
@@ -0,0 +1,24 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+/**
+ * Return a normalized string for the OS name specifically for Docker
+ * @param os pre-normalized OS name
+ * @returns normalized OS name
+ */
+export function normalizeContainerOS(os?: string): string {
+    os = os?.toLowerCase() || 'linux'; // default to linux if not specified (null/undefined or empty string)
+    switch (os) {
+        case 'windows':
+            return 'windows';
+        case 'mac':
+            return 'darwin';
+        case 'linux':
+            return 'linux';
+        default:
+            return os;
+    }
```

Comment:
A shorter way to write it with also a little extra resilience
```suggestion
    switch (os) {
        case 'mac':
        case 'macos':
            return 'darwin';
        default:
            return os;
    }
```